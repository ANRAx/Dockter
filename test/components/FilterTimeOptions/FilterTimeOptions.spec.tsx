import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FilterTimeOptions from '../../../app/components/FilterTimeOptions';

describe('FilterTimeOptions component', () => {
  // TODO: Update to match schema
  let filterOptions = {
    container_id: [],
    container_name: [],
    container_image: [],
    status: [],
    stream: [],
    timestamp: {
      from: '',
      to: '',
    },
    host_ip: [],
    host_port: [],
    log_level: [],
  };
  const setFilterOptions = (newState) => (filterOptions = newState);

  let fromTimeStamp = { date: '', time: '' };
  const setFromTimeStamp = (newTimeStamp) => (fromTimeStamp = newTimeStamp);

  let toTimeStamp = { date: '', time: '' };
  const setToTimeStamp = (newTimeStamp) => (toTimeStamp = newTimeStamp);

  it('should match exact snapshot', () => {
    const tree = renderer
      .create(
        <FilterTimeOptions
          filterOptions={filterOptions}
          setFilterOptions={setFilterOptions}
          fromTimestamp={fromTimeStamp}
          setFromTimestamp={setFromTimeStamp}
          toTimestamp={toTimeStamp}
          setToTimestamp={setToTimeStamp}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
