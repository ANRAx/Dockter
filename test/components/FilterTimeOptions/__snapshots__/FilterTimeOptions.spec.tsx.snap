// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTimeOptions component should match exact snapshot 1`] = `
<div
  className="inline border rounded-md mx-4 pl-4 py-2"
>
  <div
    className="inline pr-2 mr-2 border-r"
  >
    From:
  </div>
  <input
    id="from-date-input"
    max="2020-11-07"
    onChange={[Function]}
    type="date"
  />
  <input
    id="from-time-input"
    max="15:6"
    onChange={[Function]}
    type="time"
  />
  <div
    className="inline pr-2 mr-2 border-r"
  >
    To:
  </div>
  <input
    id="to-date-input"
    max="2020-11-07"
    onChange={[Function]}
    type="date"
  />
  <input
    id="to-time-input"
    max="15:6"
    onChange={[Function]}
    type="time"
  />
  <button
    className="font-bold px-4 ml-2 border-l"
    id="timestamp-submit"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
